librz/analysis/var.c

```diff
@@ -104,6 +105,36 @@ RZ_API void rz_analysis_var_resolve_overlaps(RzAnalysisVar *var) {
 	rz_pvector_free(cloned_vars);
 }
 
+static inline RzAnalysisVar *fcn_var_get_or_new(RzAnalysisFunction *fcn,
+	const char *name, RzAnalysisVarStorage *stor, const RzType *typ) {
+	if (!(fcn && stor && RZ_STR_ISNOTEMPTY(name))) {
+		return NULL;
+	}
+	RzAnalysisVar *existing = rz_analysis_function_get_var_byname(fcn, name);
+	if (existing && !storage_equals(&existing->storage, stor)) {
+		// var name already exists at a different kind+delta
+		return NULL;
+	}
+	RzAnalysisVar *o = rz_analysis_function_get_var_at(fcn, stor);
+	if (!o) {
+		o = RZ_NEW0(RzAnalysisVar);
+		if (!o) {
+			return NULL;
+		}
+		rz_pvector_push(&fcn->vars, o);
+		o->fcn = fcn;
+		rz_vector_init(&o->accesses, sizeof(RzAnalysisVarAccess), NULL, NULL);
+		rz_vector_init(&o->constraints, sizeof(RzTypeConstraint), NULL, NULL);
+	} else {
+		free(o->name);
+		if (o->type != typ) {
+			// only free if not assigning the own type to itself
+			rz_type_free(o->type);
+			o->type = NULL;
+		}
```

Comment:
I'm not sure I understand why you are cleaning these things. Could you explain?
Also, if you are freeing the `name` maybe it's better to set it to NULL.

Reply:
Yes, it's just move from `rz_analysis_function_set_var`